tests/routers/test_hgvs.py

```diff
@@ -79,67 +103,158 @@ def test_hgvs_genes(client, setup_router_db):
 
 
 def test_hgvs_gene_info_valid(client, setup_router_db):
-    response = client.get(f"/api/v1/hgvs/genes/{VALID_GENE}")
-    assert response.status_code == 200
-    assert response.json()['hgnc'] == VALID_GENE
-    assert response.json()['descr'] is not None
+    with requests_mock.mock() as m:
+
+        m.get(
+            f'https://cdot.cc/gene/{VALID_GENE}',
+            headers={'Content-Type': 'application/json'},
+            json={'gene_symbol': 'BRCA1', 'aliases': "BRCAI, BRCC1", "map_location": "17q21.31", "description": "BRCA1 DNA repair associated", "summary": "This gene, etc"},
+        )
+        response = client.get(f"/api/v1/hgvs/genes/{VALID_GENE}")
+
+        assert m.called
+
+        assert response.status_code == 200
+        assert response.json()['hgnc'] == VALID_GENE
+        assert response.json()['descr'] is not None
 
 
 def test_hgvs_gene_info_invalid(client, setup_router_db):
-    response = client.get(f"/api/v1/hgvs/genes/{INVALID_GENE}")
-    # XXX this probably SHOULD return a 404, but currently returns a 200 with None
-    # assert response.status_code == 404
-    assert response.status_code == 200
-    assert response.json() is None
+    with requests_mock.mock() as m:
+
+        m.get(
+            'https://cdot.cc/gene/fnord',
+            status_code=404,
+        )
+        response = client.get(f"/api/v1/hgvs/genes/{INVALID_GENE}")
+        # XXX this probably SHOULD return a 404, but currently returns a 200 with None
```

Comment:
want to file an issue, TODO(#issue) all of these, and merge as is? or is this WIP?

Reply:
I can raise -api and -ui issues for the 5 `XXX` items there ... it's not that they're *wrong* per se it's jut that a GET for a thing which doesn't exist would normally by convention return a 404.

Plenty of other APIs break this convention though so I'm okay with not worrying about it.